controllers/activemqartemis_controller_test.go

```diff
@@ -126,6 +126,266 @@ var _ = Describe("artemis controller", func() {
+
+					candidate.Spec.Acceptors = []brokerv1beta1.AcceptorType{
+						{
+							Name:               "amqp-ssl",
+							Protocols:          "amqp",
+							Port:               5671,
+							SSLEnabled:         true,
+							SSLSecret:          commonSecretName,
+							EnabledProtocols:   "TLSv1,TLSv1.1,TLSv1.2",
+							NeedClientAuth:     false,
+							WantClientAuth:     false,
+							Expose:             true,
+							AnycastPrefix:      "jms.queue.",
+							MulticastPrefix:    "/topic/",
+							ConnectionsAllowed: 5,
+						},
+					}
+
+					candidate.Spec.Connectors = []brokerv1beta1.ConnectorType{
+						{
+							Name:       "connector-ssl",
+							Host:       "0.0.0.0",
+							Type:       "tcp",
+							Port:       5671,
+							SSLEnabled: true,
+							SSLSecret:  "common-ssl-secret",
+							Expose:     true,
+						},
+					}
+				})
+
+				ssKey := types.NamespacedName{
+					Name:      namer.CrToSS(brokerCr.Name),
+					Namespace: defaultNamespace,
+				}
+
+				By("verify pod is up")
+				currentSS := &appsv1.StatefulSet{}
+				podKey := types.NamespacedName{Name: namer.CrToSS(brokerCr.Name) + "-0", Namespace: defaultNamespace}
+				pod := &corev1.Pod{}
+				Eventually(func(g Gomega) {
+					g.Expect(k8sClient.Get(ctx, ssKey, currentSS)).Should(Succeed())
+					g.Expect(k8sClient.Get(ctx, podKey, pod)).Should(Succeed())
+					g.Expect(len(pod.Status.ContainerStatuses)).Should(Equal(1))
+					g.Expect(pod.Status.ContainerStatuses[0].State.Running).Should(Not(BeNil()))
+				}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+				By("checking service is created for console")
+				serviceName := brokerCr.Name + "-amqp-ssl-0-svc"
+				service := corev1.Service{}
+				serviceKey := types.NamespacedName{
+					Name:      serviceName,
+					Namespace: defaultNamespace,
+				}
+				Eventually(func(g Gomega) {
+					g.Expect(k8sClient.Get(ctx, serviceKey, &service)).To(Succeed())
+
+				}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+				By("checking service is created for acceptor")
+				serviceName = brokerCr.Name + "-amqp-ssl-0-svc"
+				service = corev1.Service{}
+				serviceKey = types.NamespacedName{
+					Name:      serviceName,
+					Namespace: defaultNamespace,
+				}
+				Eventually(func(g Gomega) {
+					g.Expect(k8sClient.Get(ctx, serviceKey, &service)).To(Succeed())
+				}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+				By("checking service is created for connector")
+				serviceName = brokerCr.Name + "-connector-ssl-0-svc"
+				service = corev1.Service{}
+				serviceKey = types.NamespacedName{
+					Name:      serviceName,
+					Namespace: defaultNamespace,
+				}
+				Eventually(func(g Gomega) {
+					g.Expect(k8sClient.Get(ctx, serviceKey, &service)).To(Succeed())
+				}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+				if isOpenshift {
+
+					By("check route is created for console")
```

Comment:
We could just validate that the pod is ready, why do we need to validate ingress or route resources?

Reply:
I do think that's necessary to check further the ingress or routes are created. Pod is ready only means pod is up it doesn't mean we fixed the shared secret problem.